torch/nn/parallel/scatter_gather.py

```diff
@@ -1,7 +1,15 @@
 import torch
 from ._functions import Scatter, Gather
+import warnings
+
+__all__ = ['scatter', 'scatter_kwargs', 'gather']
 
 def is_namedtuple(obj):
+    # Check if type was created from collections.namedtuple or a typing.NamedTuple.
+    warnings.warn("is_namedtuple is deprecated, please use the python checks instead")
+    return _is_namedtuple(obj)
+
+def _is_namedtuple(obj):
```

Comment:
@anjali411 I think this change started throwing warnings in the nightlies when calling e.g. `gather()` or `scatter()` (defined below in this file) because they still relie on `is_namedtuple()`. Since `is_namedtuple()` is now deprecated, would it make sense to update the code-base to rely on `_is_namedtuple()`?

Reply:
@NicolasHug my bad! yeah we should update all instances in this file to use `_ is_namedtuple`